examples/acoustic/Acoustic_codegen.py

```diff
@@ -55,33 +55,93 @@ def damp_boundary(damp, nbp):
         # Initialize damp by calling the function that can precompute damping
         damp_boundary(self.damp.data, nbpml)
 
-        if len(self.damp.shape) == 2 and self.src.receiver_coords.shape[1] == 3:
-            self.src.receiver_coords = np.delete(self.src.receiver_coords, 1, 1)
-        if len(self.damp.shape) == 2 and self.data.receiver_coords.shape[1] == 3:
-            self.data.receiver_coords = np.delete(self.data.receiver_coords, 1, 1)
-
         if auto_tuning:  # auto tuning with dummy forward operator
-            fw = ForwardOperator(self.model, self.src, self.damp, self.data,
+            nt, nrec = self.data.shape
+            nsrc = self.source.shape[1]
+            ndim = len(self.damp.shape)
+            h = self.model.get_spacing()
+            dtype = self.damp.dtype
+            nbpml = self.model.nbpml
+            dt = self.model.get_critical_dt()
+            if self.t_order == 4:
+                dt *= 1.73
```

Comment:
Yes, there are many those things around but I concentrated on the essentials in this PR. I will attempt a throurgh re-design/clean-up of the examples within the next few PRs.